### PR TITLE
Parameterised supervisor port

### PR DESCRIPTION
**Pull request by devang-gaur**
fabric-hdd-bug-id: fabric8-hdd/supervisor#41
Signed-off-by: dev-gaur dgaur@redhat.com

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ var express = require("express"),
 app.use(express.json())
 app.use(express.urlencoded({ extended: true }))
 
-var port = process.env.PORT || process.env.OPENSHIFT_NODEJS_PORT || 9090,
+var port = process.env.HDD_SUPERVISOR_PORT || process.env.OPENSHIFT_NODEJS_PORT || 9090,
```
Comment (anmolbabu) [inline main.js]: Now that we are renaming this, lets keep the name consistent rename process.env.OPENSHIFT_NODEJS_PORT also as process.env.HDD_SUPERVISOR_PORT

**Review by anmolbabu — CHANGES_REQUESTED** [review]
@dev-gaur You need to also consider the changes required from template.yaml and the deploy.sh

File: main.js (hunk 1 above)
Comment (devang-gaur) [reply main.js]: Should I just omit process.env.OPENSHIFT_NODEJS_PORT ?

Comment (anmolbabu) [reply main.js]: aah yes you can omit 1
anyway both of them are from same source just 2 different env vars(from common parent source)..

Comment (devang-gaur) [reply main.js]: exactly . I made the changes in the morning itself but forgot to notify you.

**Review by anmolbabu — APPROVED** [review]
@dev-gaur Verified ?

**Comment by mik-dass** [comment]
LGTM

**Comment by devang-gaur** [comment]
Workflow:

Add your parameters to be overriden in the openshift/supervisor.env file.
Run openshift/deploy.sh

To test the PORTs, run "oc get svc/hdd-supervisor" and check the port value.
to test the route, SSH into the minishift VM, and do a curl on the route specified.

**Review by anmolbabu — APPROVED** [review]
@dev-gaur 

LGTM

But have you verified using a non-default port and observed that the app is deployed on non-default port along with service and port exposed on the same ?

I can merge this PR if you ack the PR as Verified

**Comment by devang-gaur** [comment]
Yes, I verified if.

**Comment by devang-gaur** [comment]
Yes, I verified it.

File: openshift/supervisor.env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+SUPERVISOR_PORT=
```
Comment (anmolbabu) [inline openshift/supervisor.env:1]: Provide safe defaults to mandatory params

**Review by anmolbabu — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -25,8 +25,11 @@ RUN groupadd -r node \
 ENV USER node
 USER node
 
+ARG PORT=9090
+ENV HDD_SUPERVISOR_PORT ${PORT}
```
Comment (anmolbabu) [inline Dockerfile:29]: Do you need changes to cico_setup.sh for this to be picked up even during as part of CI/CD pipelines for image build?

File: openshift/supervisor.env (hunk 2 above)
Comment (devang-gaur) [reply openshift/supervisor.env:1]: The defaults are in the template.yaml file.
This file is for parameters to override.

File: Dockerfile (hunk 3 above)
Comment (devang-gaur) [reply Dockerfile:29]: oh, I missed that.

File: openshift/supervisor.env (hunk 2 above)
Comment (anmolbabu) [reply openshift/supervisor.env:1]: I think it didn't work for me..
The deploy.sh failed without actually adding value for the parameter SUPERVISOR_PORT

File: Dockerfile (hunk 3 above)
Comment (devang-gaur) [reply Dockerfile:29]: Hey, changes made!!!!

I've append the changes to infra_test script too.

**Review by anmolbabu — APPROVED** [review]
LGTM

**Pull request merged**